core/tests/puppeteer-acceptance-tests/puppeteer-testing-utilities/user-factory.ts

```diff
@@ -138,10 +142,15 @@ export class UserFactory {
     username: string,
     email: string,
     roles: OptionalRoles<TRoles> = [] as OptionalRoles<TRoles>
-  ): Promise<LoggedInUser & MultipleRoleIntersection<TRoles>> {
-    let user = UserFactory.composeUserWithRoles(BaseUserFactory(), [
-      LoggedInUserFactory(),
-    ]);
+  ): Promise<
+    LoggedInUser & ExplorationEditor & MultipleRoleIntersection<TRoles>
+  > {
+    let user = UserFactory.composeUserWithRoles(
```

Comment:
I don't understand why you need a nested composeUserWithRoles() here. /cc @jnvtnguyen

Reply:
Actually, true, the nested composeUserWithRoles here is unnecessary as we can assign multiple roles in a single call. I've updates it to call composeUserWithRoles once with both LoggedInUserFactory and ExplorationEditorFactory passed in the roles array.

Thanks.